### ISSUE TITLE
MOSEK: one keyword argument for all continuous accuracy parameters

### ISSUE THREAD

**Issue by rileyjmurray** [cvxpy/cvxpy#1206]
This is a follow-up to the comments on #890. For one reason or another, certain parameters passed in ``mosek_params`` produce unexpected behavior. Maybe the behavior in question is not a bug, and I'm just misunderstanding these parameters. Maybe there is a bug in CVXPY or MOSEK's python interface. Paging @aszekMosek, so you can investigate when you have the time. Also paging  @iarbel84 since you have an interest in specifying termination criteria.

Based on the following small example I know that we are correctly passing some parameters to MOSEK (specifically, ``"MSK_IPAR_INTPNT_MAX_ITERATIONS"`` is being respected). However, the precision parameters aren't having the expected effect on MOSEK's solution status.
```
n = 10
m = 4
np.random.seed(0)
A = np.random.randn(m, n)
x = np.random.randn(n)
y = A.dot(x)

# Solve a simple basis pursuit problem for testing purposes.
z = cp.Variable(n)
objective = cp.Minimize(cp.norm1(z))
constraints = [A @ z == y]
problem = cp.Problem(objective, constraints)

mosek_params = {
    #mosek.dparam.basis_tol_x: 1e-8,
    "MSK_DPAR_INTPNT_CO_TOL_DFEAS": 1e-5,
    "MSK_DPAR_INTPNT_CO_TOL_PFEAS": 1e-5,
    "MSK_DPAR_INTPNT_CO_TOL_REL_GAP": 1e-5,
    "MSK_IPAR_INTPNT_MAX_ITERATIONS": 5
}
problem.solve(solver=cp.MOSEK, mosek_params=mosek_params, verbose=True)
```
With the problem data below on CVXPY 1.1.7 and MOSEK 9.2.18, I get the following output. Note that at iteration 5, all residuals are within the ``1e-5`` tolerance specified in ``mosek_params``. Despite this, MOSEK returns an "unknown" solution status.
```
Problem
  Name                   :                 
  Objective sense        : max             
  Type                   : LO (linear optimization problem)
  Constraints            : 20              
  Cones                  : 0               
  Scalar variables       : 24              
  Matrix variables       : 0               
  Integer variables      : 0               

Optimizer started.
Problem
  Name                   :                 
  Objective sense        : max             
  Type                   : LO (linear optimization problem)
  Constraints            : 20              
  Cones                  : 0               
  Scalar variables       : 24              
  Matrix variables       : 0               
  Integer variables      : 0               

Optimizer  - threads                : 4               
Optimizer  - solved problem         : the dual        
Optimizer  - Constraints            : 4
Optimizer  - Cones                  : 0
Optimizer  - Scalar variables       : 20                conic                  : 0               
Optimizer  - Semi-definite variables: 0                 scalarized             : 0               
Factor     - setup time             : 0.00              dense det. time        : 0.00            
Factor     - ML order time          : 0.00              GP order time          : 0.00            
Factor     - nonzeros before factor : 10                after factor           : 10              
Factor     - dense dim.             : 0                 flops                  : 1.78e+02        
ITE PFEAS    DFEAS    GFEAS    PRSTATUS   POBJ              DOBJ              MU       TIME  
0   3.6e+00  1.4e+01  6.5e+01  0.00e+00   2.536797490e+00   1.974590736e+01   1.3e+01  0.00  
1   9.7e-01  3.7e+00  1.8e+01  7.69e-01   5.465345451e+00   1.077293483e+01   3.5e+00  0.00  
2   1.3e-01  4.9e-01  2.3e+00  1.02e+00   5.241293716e+00   5.932521242e+00   4.6e-01  0.00  
3   2.3e-02  8.9e-02  4.2e-01  9.74e-01   5.345089253e+00   5.472270581e+00   8.3e-02  0.00  
4   1.7e-03  6.6e-03  3.1e-02  9.96e-01   5.341809858e+00   5.351277713e+00   6.2e-03  0.00  
5   3.3e-07  1.3e-06  6.0e-06  1.00e+00   5.343117088e+00   5.343118928e+00   1.2e-06  0.00  
Optimizer terminated. Time: 0.00    


Interior-point solution summary
  Problem status  : UNKNOWN
  Solution status : UNKNOWN
  Primal.  obj: 5.3431170880e+00    nrm: 1e+00    Viol.  con: 7e-08    var: 2e-08  
  Dual.    obj: 5.3431189278e+00    nrm: 5e+00    Viol.  con: 0e+00    var: 5e-07  
```

**Comment by rileyjmurray** [comment]
@aszekMosek I can confirm that this still happens with MOSEK 9.3.10 and CVXPY 1.1.17. Can you take a look and let us know if you think there might be a mistake on CVXPY's side?

**Comment by aszekMosek** [comment]
Sorry, I forgot that somehow.

Since this is a linear problem the parameter names to use are

```
eps = 1

mosek_params = {
    #mosek.dparam.basis_tol_x: 1e-8,
    "MSK_DPAR_INTPNT_TOL_DFEAS": eps,
    "MSK_DPAR_INTPNT_TOL_PFEAS": eps,
    "MSK_DPAR_INTPNT_TOL_REL_GAP": eps,
    #"MSK_IPAR_INTPNT_MAX_ITERATIONS": 5
}
problem.solve(solver=cp.MOSEK, mosek_params=mosek_params, verbose=True)
```

For historical reasons the parameters without ``CO`` refer to the interior-point optimizer for LPs and those with ``CO`` to the interior-point conic optimizer. This may get unified at some point if those two code paths are merged but for now it s like that.

If you run with my parameters above you get a fairly inaccurate solution declared as optimal, ie. as expected.

**Comment by rileyjmurray** [comment]
Ah! Thanks so much for the reply, @aszekMosek. What do you think about letting a user provide a keyword argument ``eps`` when calling ``prob.solve(solver='MOSEK', eps=eps)``, where we set all continuous-problem termination criteria to use accuracy ``eps``? I think that would be much easier for cvxpy users, since many don't know when their problem will be compiled into an LP versus some other type of cone program.

**Comment by Paulnkk** [comment]
@rileyjmurray @aszekMosek I am currently taking a look at the issue and if I understand it correctly (please correct me if I am wrong) but "eps" can and should just be passed to solve() if the user define "mosek_params" which refer to termination criteria (for instance MSK_DPAR_INTPNT_CO_TOL_MU_RED). Here I am thinking about including all the Mosek parameters that refer to termination and require a tolerance but with this approach, there may be a drawback to update those fixed parameters every time they change in Mosek itself.

**Comment by phschiele** [comment]
Resolved via #2119.
